drivers/lora/lora_basics_modem/lbm_common.h

```diff
@@ -105,3 +120,40 @@ static inline int lbm_optional_gpio_set_dt(const struct gpio_dt_spec *spec, int
 
 /* Common LBM implementation of the LoRa API */
 extern const struct lora_driver_api lbm_lora_api;
+
+/**
+ * @brief Add a GPIO callback for DIO1 interrupts
+ *
+ * This function registers a user callback that will be invoked from interrupt
+ * context when the DIO1 interrupt fires. The user must provide a pre-allocated
+ * and initialized (except for pin mask) gpio_callback structure.
+ *
+ * Note: The driver will configure the pin mask of the callback structure to
```

Comment:
Rather than making this function do something odd (half initializing the struct), I think it would be cleaner to have `lbm_driver_add_dio1_gpio_callback` handle all of the initialisation of `struct gpio_callback`, which would only require passing the callback pointer as a third argument.